### ISSUE TITLE
Template Options

### ISSUE THREAD

**Issue by adligo** [vuejs/vue#10169]
### What problem does this feature solve?
Beat React in Benchmarks & Improve Flexibility of Template System Integration

### What does the proposed API look like?
Hi,
  I have an idea for a new template system that I would like to integrate into Vue.  The basic idea is to create a template that can be compiled / aggressively inlined into multiple languages (i.e. Java, PHP, Python, Javascript), split up into files (with chunks of HTML) for maximum flexibility, or runtime compiled.  

**I am NOT looking to add this to the Vue project but simply integrate with the Vue project.**

 **In a nutshell the templates look like this (i.e. html chunks in xml files);**

```
<?xml version="1.0" encoding="UTF-8"?>
<_templates>
<_page>
<![CDATA[<!DOCTYPE html>]]>
<html><head></head><body><div _attributeVariable="">[[contentVariable]]</div></body>
</html>
</_page>
<_content p="html.contentVariable">Hello World!</_content> 
</_templates>
```

**The Plan for this is far from complete, but could it integrate to Vue?  If so how to go about doing it?**
